packages/anoncreds/src/repository/AnonCredsRevocationRegistryDefinitionPrivateRecord.ts

```diff
@@ -0,0 +1,62 @@
+import type { TagsBase } from '@aries-framework/core'
+
+import { BaseRecord, utils } from '@aries-framework/core'
+
+export enum RevocationRegistryState {
+  Created = 'created',
+  Active = 'active',
+  Full = 'full',
+}
+
+export interface AnonCredsRevocationRegistryDefinitionPrivateRecordProps {
+  id?: string
+  revocationRegistryDefinitionId: string
+  credentialDefinitionId: string
+  value: Record<string, unknown>
+  index?: number
```

Comment:
I'm thinking whether we should add an option to manage the index that will be used for issuance outside of the framework. As mentioned before, we can have issues when multiple issuance processed are in process as they will overrwrite the value (can't do index += 1).

For us we would thus like to manage this outside of AFJ and store the index in a service where it is possible to prevent overwriting of the indexes.

Reply:
Yeah, this can be problematic in several cases, especially in batch issuance (like I think it is also for batch revocation). 

I think for now the best would be to not store any currentRevocationIndex in the record and just leave that responsibility and logic to the controller.

Reply:
The challenge for that approach is to see where to add that binding: a simple way would be setting it at `CredentialsApi.proposeCredential` step (add revocationIndex to `AnonCredsProposeCredentialFormat` and store it in credential exchange metadata). However, this will assign an index even if a credential has not actually been issued.

So we can also do it at `CredentialsApi.acceptRequest` (adding it to `AnonCredsAcceptRequestFormat`), although this would not work if ` shouldAutoRespondToRequest` happens to be true.

Another more complex option would be creating an interface for a "RevocationService" that feeds to AFJ's `AnonCredsCredentialFormatService.acceptRequest()` the proper index for a given revocationRegistryDefinitionId. That service should also decide which revocation registry is the active one instead of querying it into AFJ storage.

Now I think about it, a controller can also provide the `revocationRegistryDefinitionId` to proposeCredental/acceptRequest options, so it can manage revocation registry definition generation in a complete separate way (could be useful considering the processing time it takes to generate a new one).

Reply:
I think both upfront and callback/interface based would work.

If auto issuance is enabled we would need callback/interface based. So maybe that's the way to go? Auto issuance shouldn't be encouraged though i think.

I think if we have auto acceptance and no explicit offer we should jist throw an error if the vred def needs revocation?

Lot's of complexities -- i would like the simplest most extendable way (whatever that is 😬)

Reply:
> I think if we have auto acceptance and no explicit offer we should jist throw an error if the vred def needs revocation?

For the moment I left it like that: only manual flow is supported (which will be the most common anyway) and offerCredential/acceptProposal API receive both revocationRegistryDefinitionId and revocation index